File: sparkTtrain/src/main/scala/com/fengli/spark/project/dao/CourseClickCountDAO.scala
package com.fengli.spark.project.dao

import com.fengli.spark.project.domain.CourseClickCount
import com.fengli.spark.project.utils.HBaseUtils
import org.apache.hadoop.hbase.client.Get
import org.apache.hadoop.hbase.util.Bytes

import scala.collection.mutable.ListBuffer

/**
 * 实战课程点击数 DAO
 */
object CourseClickCountDAO {
  val tableName = "imooc_course_clickcount2"
  val cf = "info"
  val qualifer = "click_count"


  /**
   * 保存数据到HBase
   * @param list
   */
  def save(list: ListBuffer[CourseClickCount]): Unit = {

    val table = HBaseUtils.getInstance().getHTable(tableName)

    for (ele <- list){
      table.incrementColumnValue(Bytes.toBytes(ele.day_course)
      ,Bytes.toBytes(cf)
      ,Bytes.toBytes(qualifer)
      ,ele.click_count)
    }
  }


  /**
   * 根据Rowkey查值
   * @param day_course
   */
  def find(day_course: String): Long ={
    val table = HBaseUtils.getInstance().getHTable(tableName)

    val get = new Get(Bytes.toBytes(day_course))

    val value = table.get(get).getValue(cf.getBytes()
      , qualifer.getBytes())

    if (value == null) {
      0L
    }else{
      Bytes.toLong(value)
    }
  }

  def main(args: Array[String]): Unit = {
   val list = new ListBuffer[CourseClickCount]
    list.append(CourseClickCount("2018111_8", 8)
      , CourseClickCount("2018111_8", 217921)
      , CourseClickCount("2018111_9", 78978)
      , CourseClickCount("2018111_10", 87)
      , CourseClickCount("2018111_11", 111)
    )

//    save(list)

    println(find(("2018111_8"))
      , find(("2018111_8"))
      , find(("2018111_9"))
      , find(("2018111_10"))
      , find(("2018111_11"))
    )
  }
}

File: TestMaven/src/main/java/com/fengli/dao/CourseClickCountDAO.java
package com.fengli.dao;

import com.fengli.domain.CourseClickCount;
import com.fengli.utils.HBaseUtils;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * 实战课程访问数据访问层
 * @author Administrator
 */
@Component
public class CourseClickCountDAO {

    /**
     * 根据天查询
     * @param day
     * @return
     */
    public List<CourseClickCount> query(String day) throws IOException {
        List<CourseClickCount> list = new ArrayList<CourseClickCount>();

        // HBase表中根据天获取对应的访问量
        Map<String, Long> map = HBaseUtils.getInstance().query("imooc_course_clickcount2", day);

        for (Map.Entry<String, Long> entry:map.entrySet()){
            CourseClickCount model = new CourseClickCount();
            model.setName(entry.getKey());
            model.setValue(entry.getValue());
            list.add(model);
        }
        return list;
    }

    public static void main(String[] args) throws IOException {
        CourseClickCountDAO courseClickCountDAO = new CourseClickCountDAO();
        List<CourseClickCount> list = courseClickCountDAO.query("20200210");

        for (CourseClickCount c :
                list) {
            System.out.println(c.getName() + ":" + c.getValue());
        }
    }
}
